pkg/maps/metricsmap/metricsmap.go

```diff
@@ -200,49 +195,47 @@ func SyncMetricsMap() error {
 	return nil
 }
 
-// calculateNumCpus replicates the bpf linux helper equivalent `bpf_num_possible_cpus`
-// to find total number of possible CPUs i.e CPUs that have been allocated
-// resources and can be brought online if they are present.
-// Reference bpf_num_possible_cpus from
-// https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf.git/tree/tools/testing/selftests/bpf/bpf_util.h
-func calculateNumCpus() {
-	var start, end int
+// getNumPossibleCPUs returns a total number of possible CPUS, i.e. CPUs that
+// have been allocated resources and can be brought online if they are present.
+// The number is retrieved by parsing /sys/device/system/cpu/possible.
+//
+// See https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/cpumask.h?h=v4.19#n50
+// for more details.
+func getNumPossibleCPUs() int {
+	return getNumPossibleCPUsFromFile("/sys/devices/system/cpu/possible")
+}
 
-	file, err := os.Open("/sys/devices/system/cpu/possible")
+func getNumPossibleCPUsFromFile(path string) int {
+	out, err := ioutil.ReadFile(path)
 	if err != nil {
-		log.WithError(err).Error("unable to open sysfs to get CPU count")
+		log.WithError(err).Error("unable to read sysfs to get CPU count")
+		return 0
 	}
-	defer file.Close()
 
-	data := make([]byte, possibleCPUsFileLength)
-	for {
-		_, err := file.Read(data)
-		if err != nil {
-			if err == io.EOF {
-				break
-			}
-			log.WithError(err).Error("unable to read sysfs to get CPU count")
-		}
-		n, err := fmt.Sscanf(string(data), "%d-%d", &start, &end)
-		if err != nil {
-			log.WithError(err).Error("unable to parse sysfs to get CPU count")
-		}
-		if n == 0 {
+	var start, end int
+	count := 0
+	for _, s := range strings.Split(string(out), ",") {
+		// Go's scanf will return an error if a format cannot be fully matched.
+		// So, just ignore it, as a partial match (e.g. when there is only one
+		// CPU) is expected.
+		n, _ := fmt.Sscanf(s, "%d-%d", &start, &end)
+
+		switch n {
+		case 0:
 			log.WithError(err).Error("failed to retrieve number of possible CPUs!")
-		} else if n == 1 {
-			end = start
+			return 0
```

Comment:
While you're looking at this code, do you think we should include 's' in the error message above? This code is unlikely to trigger but it'd be nice to know what the input was if it is ever triggered.

Reply:
Good idea, sure!